extensions/spring-data-jpa/runtime/src/main/java/io/quarkus/spring/data/runtime/TypesConverter.java

```diff
@@ -0,0 +1,40 @@
+package io.quarkus.spring.data.runtime;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import io.quarkus.panache.common.Sort;
+
+public final class TypesConverter {
+
+    private TypesConverter() {
+    }
+
+    public static io.quarkus.panache.common.Sort toPanacheSort(org.springframework.data.domain.Sort sort) {
+        List<org.springframework.data.domain.Sort.Order> orders = sort.get().collect(Collectors.toList());
+        if (orders.isEmpty()) {
+            return null;
+        }
+
+        org.springframework.data.domain.Sort.Order firstOrder = orders.get(0);
+        Sort result = Sort.by(firstOrder.getProperty(), getDirection(firstOrder));
+        if (orders.size() == 1) {
+            return result;
+        }
+
+        for (int i = 1; i < orders.size(); i++) {
+            org.springframework.data.domain.Sort.Order order = orders.get(i);
+            result = result.and(order.getProperty(), getDirection(order));
+        }
+        return result;
+    }
+
+    private static Sort.Direction getDirection(org.springframework.data.domain.Sort.Order firstOrder) {
```

Comment:
I would rename the parameter to `order`. The parameter name is a bit misleading (probably due to an automatic refactor).